### PR TITLE
Add Menu Buttons Component

### PR DESCRIPTION
**Pull request by RyanUlch**
Simple component that does not handle state, and only affects gameplay through callbacks passed through props to open up specific modals.

### CODE REVIEW

File: src/components/MenuButtons.vue [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+			modalHandler: () => void;
+		}[];
+	}>();
+</script>
+
+<template>
+	<div class="menuBtnContainer">
+		<button
+			v-for="(btn, btnIndex) in props.btnInfoArray"
+			:key="`btn-${btnIndex}`"
```
Comment (RyanUlch) [inline src/components/MenuButtons.vue:17]: Not needed 'btn-' only needs to be unique to the iteration

File: src/components/MenuButtons.vue [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+			:title="btn.tooltip"
+			@click.prevent="btn.modalHandler"></button>
+	</div>
+</template>
+
+<style scoped>
+	/* Based on current design, buttons are in a 2x2 grid, can change grid-template-columns for adding/removing buttons */
+	.menuBtnContainer {
+		display: inline-grid;
+		grid-template-columns: 1fr 1fr;
```
Comment (RyanUlch) [inline src/components/MenuButtons.vue]: Use v-bind to dynamically create an appropriate amount of columns - probably with flex.

**Review by RyanUlch — COMMENTED** [review]
Finished Review - a couple of changes need to be made.

File: src/components/MenuButtons.vue (hunk 2 above)
Comment (RyanUlch) [reply src/components/MenuButtons.vue]: Not using v-bind here. There was a simple css solution. using calc(),

**Review by ericsartor — APPROVED** [review]

**Pull request merged**